Adapt Java bindings to latest changes in Explicit Disclosure

tudor-da opened the pull request:
This PR adapts the Java bindings to the recent changes in the Explicit Disclosure Ledger API interfaces ([source](https://github.com/digital-asset/daml/commit/25ae8d2281a8f34473dd9e3ba9d11706b83eb64f))

<!--
# Pull Request Checklist

- Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- Include appropriate tests
- Set a descriptive title and thorough description
- Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- Normal production system change, include purpose of change in description
- If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.
-->

raphael-speyer-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:
```diff
@@ -44,16 +53,74 @@ public TransactionFilterOuterClass.InterfaceFilter toProto(Identifier interfaceI
+    }
+  }
+
+  public static final class Template {
```
Also probably better as an enum.

```suggestion
  public static enum Template {
    INCLUDE_PAYLOAD(true),
    HIDE_PAYLOAD(false);
```

tudor-da replied:
Done

mziolekda commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:
```diff
@@ -18,22 +19,30 @@ public static Filter fromProto(TransactionFilterOuterClass.Filters filters) {
 
-    public static final Interface INCLUDE_VIEW = new Interface(true);
-    public static final Interface HIDE_VIEW = new Interface(false);
+    public static final Interface HIDE_VIEW_HIDE_PAYLOAD = new Interface(false, false);
```
the rename to `created_event_blob` is about to hit the repo, so then these fields should follow suit.

tudor-da replied:
I've went the verbose way in the end for expressiveness (`*_CREATED_EVENT_BLOB`)

tudor-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DisclosedContract.java:
```diff
@@ -5,27 +5,46 @@
-  public final Arguments arguments;
-  public final ContractMetadata contractMetadata;
+  public final Optional<Arguments> arguments;
+  public final Optional<ContractMetadata> contractMetadata;
```
Yes, proceeded as suggested, thanks!

tudor-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:
```diff
@@ -18,22 +19,30 @@ public static Filter fromProto(TransactionFilterOuterClass.Filters filters) {
+   * {@link #HIDE_VIEW_HIDE_PAYLOAD} and {@link #INCLUDE_VIEW_HIDE_PAYLOAD} and {@link
+   * #HIDE_VIEW_INCLUDE_PAYLOAD} and {@link #INCLUDE_VIEW_INCLUDE_PAYLOAD}.
    */
   public static final class Interface {
```
Done

tudor-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java:
```diff
@@ -70,6 +75,72 @@ public CreatedEvent(
+   *     deprecated. Use the constructor with {@link #createEventPayload} instead. Since Daml 2.8.0
+   */
+  @Deprecated
+  public CreatedEvent(
```
Indeed, went without backwards compatibility support here (see ad242ec8d8c679ac62f12eb10a0759b5790b7d3a)

tudor-da commented:
@raphael-speyer-da Thanks for the review. I've addressed the review comments but generally I've changed some DTO's to non-backwards compatibility (`CreatedEvent` and `DisclosedContract`) since the affected fields haven't indeed been released in a GA feature. Please re-review

raphael-speyer-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:
```diff
@@ -18,22 +18,33 @@ public static Filter fromProto(TransactionFilterOuterClass.Filters filters) {
-  public static final class Interface {
-    public final boolean includeInterfaceView;
-    // add equals and hashCode if adding more fields
+  public enum Interface {
```
These types should probably be `static`.

```suggestion
  public static enum Interface {
```

tudor-da replied:
IJ was suggesting that is not necessary for inner enums. But fine by me to apply

raphael-speyer-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:
```diff
@@ -44,16 +55,43 @@ public TransactionFilterOuterClass.InterfaceFilter toProto(Identifier interfaceI
+    }
+  }
+
+  public enum Template {
```
```suggestion
  public static enum Template {
```

raphael-speyer-da reviewed (approved):
LGTM modulo comments

tudor-da commented on language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/InclusiveFilter.java:
```diff
@@ -28,15 +45,23 @@ public InclusiveFilter(@NonNull Set<@NonNull Identifier> templateIds) {
   }
 
+  /**
+   * @deprecated Use the constructor without with {@link #templateFilters} instead IDs. Since Daml
```
Apologies, applied but didn't pull after 🤦
